ctp2_code/gs/gameobj/CityData.cpp

```diff
@@ -5974,6 +5973,15 @@ void CityData::MakeFranchise(sint32 player)
 	m_franchiseTurnsRemaining = -1;
 }
 
+void CityData::RemoveFranchise()
+{
+	if(m_franchise_owner < 0) // according to CityData::Unconvert
```

Comment:
@MartinGuehmann  is it critical here to return if `m_franchise_owner < 0`?

Reply:
That depends, if whatever follows would be a lot of code then yes. However, here you just want to reset the franchise and if you want to do it cleanly then both members have to set to -1.

I think in the end it is a speed issue. So if the code is used very often, then you would optimize it in such a way, but then the function should be inline. However, I do not expect that this code is heavily used and so it can be just executed.